### ISSUE TITLE
USWDS - Bug: Radio hints are inside form label

### ISSUE THREAD

**Issue by jaredcunha** [uswds/uswds#6525]
### Describe the bug

The hint to text for radio buttons is stuffed inside the label, creating names for the labels that are longer than necessary. 

### Steps to reproduce the bug

1. Using a screen reader, interact with the USWDS tiled radio buttons using a screenreader, preferably NVDA or JAWS
2. When the radio option within selected JAWS announced the label AND the description followed by "radio button checked"

On the USWDS site, the announcement is:

> Frederick Douglass This is optional text that can be used to describe the label in more. Radio button checked. 2 of 4

### Expected Behavior

The descriptions should be distinguishable from the input names. 

Using JAWS on the USWDS, it should be 

> Frederick Douglass. Radio button checked. 2 of 4.  This is optional text that can be used to describe the label in more." 

In this announcement, it is more clear what text is the name of the input and what is the description.

### Related code

The current code is this:
```html
<div class="usa-radio">
      <input class="usa-radio__input usa-radio__input--tile" id="historical-douglass-2" type="radio" name="historical-figures-2" value="frederick-douglass">
      <label class="usa-radio__label" for="historical-douglass-2">Frederick Douglass
        <span class="usa-checkbox__label-description">This is optional text that can be used to describe the label in more
          detail.</span></label>
</div>
```

Suggested fix is this (see [codepen](https://codepen.io/jaredcunha/pen/yyYGpwr?editors=1100) example):
```html
<div class="usa-radio">
        <input class="usa-radio__input usa-radio__input--tile" id="historical-douglass-2" type="radio" name="historical-figures-2" value="frederick-douglass" aria-describedby="desc">
        <label class="usa-radio__label" for="historical-douglass-2">Frederick Douglass
          <span id="desc" aria-hidden="true" class="usa-checkbox__label-description">This is optional text that can be used to describe the label in more
            detail.</span></label>
</div>
```

- The suggestion adds `aria-hidden` and `id` to the hint element
- The suggestion adds `aria-describedby` to associate the input with the hint element

### Screenshots

**This is the current implementation**
<img width="815" height="166" alt="Image" src="https://github.com/user-attachments/assets/15be335c-093b-4edb-b7d6-fba2b022093b" />

**The suggested is announced like this.**
<img width="495" height="248" alt="Image" src="https://github.com/user-attachments/assets/64a17e45-c3d7-4e72-a836-76a58b36fd04" />

### Additional context

Using VoiceOver, there isn't much of a distinction between the current implementation and the suggested implementation. But those distinctions are much more apparent using NVDA or Jaws.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.
